packages/pcd-collection/src/PCDCollection.ts

```diff
@@ -43,6 +53,142 @@ export class PCDCollection {
     this.recalculateAndEmitHash();
   }
 
+  public async tryExec(
+    action: PCDAction,
+    permissions: PCDPermission[]
+  ): Promise<boolean> {
+    for (const permission of permissions) {
+      try {
+        const result = await this.tryExecutingActionWithPermission(
+          action,
+          permission
+        );
+
+        if (result) {
+          return true;
+        }
+      } catch (e) {
+        console.log(e);
```

Comment:
how do we report errors? i think here it is actually pretty bad to `console.log` because there are `m`x`n` combinations of actions and permissions, only `m` of which will work, causing way more logs than necessary.

Reply:
I've refactored this so we don't loop over all permissions. Exceptions here are rare, only occurring if we're trying to add the same PCD twice, or to multiple folders. Regular permission problems are not exceptions.